can/interfaces/vector/canlib.py

```diff
@@ -196,22 +263,52 @@ def recv(self, timeout=None):
                 time.sleep(self.poll_interval)
 
     def send(self, msg, timeout=None):
-        message_count = ctypes.c_uint(1)
         msg_id = msg.arbitration_id
+
         if msg.id_type:
             msg_id |= vxlapi.XL_CAN_EXT_MSG_ID
+
         flags = 0
-        if msg.is_remote_frame:
-            flags |= vxlapi.XL_CAN_MSG_FLAG_REMOTE_FRAME
-        xl_event = vxlapi.XLevent()
-        xl_event.tag = vxlapi.XL_TRANSMIT_MSG
-        xl_event.tagData.msg.id = msg_id
-        xl_event.tagData.msg.dlc = msg.dlc
-        xl_event.tagData.msg.flags = flags
-        for idx, value in enumerate(msg.data):
-            xl_event.tagData.msg.data[idx] = value
-        vxlapi.xlCanTransmit(self.port_handle, self.mask, message_count, xl_event)
 
+        if self.fd:
+            if msg.dlc > vxlapi.MAX_MSG_LEN: #extended data length (XL_CAN_RXMSG_FLAG_EDL) not in msg class
+                flags |= vxlapi.XL_CAN_TXMSG_FLAG_EDL
+            if msg.bitrate_switch:
+                flags |= vxlapi.XL_CAN_TXMSG_FLAG_BRS
+            if msg.is_remote_frame:
+                flags |= vxlapi.XL_CAN_TXMSG_FLAG_RTR
+                        
+            message_count = 1
+            MsgCntSent = ctypes.c_uint(1)
+            
+            XLcanTxEvent = vxlapi.XLcanTxEvent()
+            XLcanTxEvent.tag = vxlapi.XL_CAN_EV_TAG_TX_MSG
+            XLcanTxEvent.transId = 0xffff
+            
+            XLcanTxEvent.tagData.canMsg.canId = msg_id
+            XLcanTxEvent.tagData.canMsg.msgFlags = flags
+            XLcanTxEvent.tagData.canMsg.dlc = msg.dlc
```

Comment:
Use `can.util.len2dlc()` since I believe this should be the code, not the number of bytes.